astropy/modeling/core.py

```diff
@@ -1061,13 +1061,7 @@ def _validate_input_shapes(self, inputs, argnames, model_set_axis):
                 )
             )
 
-        input_shape = check_broadcast(*all_shapes)
-        if input_shape is None:
-            raise ValueError(
-                "All inputs must have identical shapes or must be scalars."
-            )
-
-        return input_shape
+        return check_broadcast(*all_shapes)
```

Comment:
After examining the `IncompatableShapeError` I think I like previous error message better.

For  example the `test_core` test which was refactored gives the message:
```
astropy.utils.shapes.IncompatibleShapeError: ((2, 2), 0, (3, 2), 1)
```
which is a bit opaque to as to what the issue is.

Reply:
Alright. How about if I raise a new IncompatibleShapeError but with a better explanation?

Reply:
The current message is actually pretty clear about what users should be doing.

Reply:
OK, got it. That will make this an API change, though, since `IncompatibleShapeError` doesn't support custom error messages, but `ValueError` does. Sorry, @pllim!

Reply:
No worries. Let me know when to update the labels again. 😆  API change needs a change log for sure.